kvpress/presses/base_press.py

```diff
@@ -178,16 +178,17 @@ def __call__(self, model: PreTrainedModel) -> Generator:
         if not isinstance(model, SUPPORTED_MODELS):
             logger.warning(f"Model {type(model)} not tested, supported models: {SUPPORTED_MODELS}")
 
-        if isinstance(model, Gemma3ForCausalLM):
-            logger.warning("Compression in Gemma3 is only applied to layer without sliding window attention")
+        if isinstance(model, Gemma3ForConditionalGeneration):
+            logger.warning_once("Compression in Gemma3 is only applied to layer without sliding window attention")
 
         hooks = []
         try:
-            for layer in model.model.layers:
-                if isinstance(model, Gemma3ForCausalLM) and layer.is_sliding:
+            lm = model.model.language_model if isinstance(model, Gemma3ForConditionalGeneration) else model.model
```

Comment:
You could make this more general, i.e. `if hasattr(model.model, "language_model")`.
Also, `if getattr(layer.self_attn, "is_sliding", False)` could be used to potentially support other models with the same naming convention (maybe add a comment that it is for gemma3).